src/libcore/char.rs

```diff
@@ -377,7 +391,16 @@ impl Iterator for EscapeDefault {
                 Some(c)
             }
             EscapeDefaultState::Done => None,
-            EscapeDefaultState::Unicode(ref mut iter) => iter.next()
+            EscapeDefaultState::Unicode(ref mut iter) => iter.next(),
+        }
+    }
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        match self.state {
+            EscapeDefaultState::Char(_) => (1, Some(1)),
+            EscapeDefaultState::Backslash(_) => (2, Some(2)),
+            EscapeDefaultState::Unicode(ref iter) => iter.size_hint(),
```

Comment:
I actually tested this with `(0, Some(10))` before adding `size_hint` to `EscapeUnicode` and using it here, didn't noticed any significant difference, so I guess slowdown comes from iterator construction. I'll double check that though.